packages/manager/apps/telecom/src/app/telecom/pack/move/offers/move-offers.controller.js

```diff
@@ -239,32 +240,70 @@ export default class PackMoveOffersCtrl {
       ) {
         if (option.name !== optionName) {
           set(option, 'selected', false);
+        } else {
+          set(option, 'selected', true);
         }
         if (optionName.match(/^gtr_\d{1,2}m_/) && option.selected === true) {
           optionComfort = true;
         }
         set(offer, 'gtrComfortActivated', optionComfort);
       }
+      if (optionName === 'none') {
+        set(option, 'selected', false);
+      }
     });
   }
 
-  updateOfferPriceAndGtr(value, offer, optionName) {
-    this.constructor.updateSelectedGtrOption(offer, optionName);
-    this.constructor.updateOfferDisplayedPrice(value, offer, optionName);
+  updateOfferPriceAndGtr(offer) {
+    this.constructor.updateSelectedGtrOption(offer, offer.gtrSelected);
+    this.constructor.updateOfferDisplayedPrice(1, offer, offer.gtrSelected);
   }
 
   static isChosen(option) {
     return option.choosedValue > 0;
   }
 
-  static isGtrOption(optionName) {
-    return optionName.startsWith('gtr_');
-  }
-
   static isGtrOptionExist(offer) {
     return offer.options.some((option) => option.name.startsWith('gtr_'));
   }
 
+  initGtrOptions() {
+    this.offers.forEach((offer) => {
+      // Initialize GTR options with a none value
+      let gtrOptions = [
+        {
+          label: this.$translate.instant('pack_move_no_gtr'),
+          name: 'none',
+        },
+      ];
+
+      const optionsMap = Object.values(offer.options)
+        .map((option) => {
+          if (option.name.startsWith('gtr_')) {
+            const gtrOption = option;
+            gtrOption.label = this.$translate.instant(
+              `telecom_pack_migration_${gtrOption.name}`,
+              { price: gtrOption.optionalPrice.text },
+            );
+            return gtrOption;
+          }
+          return undefined;
+        })
+        .filter((el) => el !== undefined);
+
+      if (optionsMap.length > 0) {
+        gtrOptions = gtrOptions.concat(optionsMap);
+      } else {
+        gtrOptions = [];
+      }
+
+      if (gtrOptions.length === 1) {
+        gtrOptions = [];
+      }
+      set(offer, 'gtrOptions', gtrOptions);
+    });
+  }
```

Comment:
**nitpick:** 

1. Inappropriate usage of `Array.prototype.forEach` to modify the items in the array
2. Usage of `Array.prototype.map` and `Array.prototype.filter` can be reversed to have better readability
3. Unnecessary local variables can be avoided

```suggestion
  initializeGtrOptions() {
    this.process.migrationOffers.result.offers = this.process.migrationOffers.result.offers.map((offer) => {
      const gtrOptions = Object.values(offer.options)
        .filter((option) => option.name.startsWith('gtr_'))
        .map((option) => ({
            ...option,
            label: this.$translate.instant(
              `telecom_pack_migration_${gtrOption.name}`,
              { price: gtrOption.optionalPrice.text },
            ),
        }));

      if (gtrOptions.length > 0) {
        // Initialize GTR options with a none value
        gtrOptions.unshift({
          name: this.GTR_NONE,
          label: this.$translate.instant('telecom_pack_migration_no_gtr'),
        });
      }
      return {
        ...offer,
        gtrOptions,
      };
    });
  }
```